models/image.py

```diff
@@ -905,44 +714,70 @@ def from_ref_and_new(cls, ref, new):
 
         Parameters
         ----------
-        ref: Image object
-            The reference image to use.
-        new: Image object
+        new_image: Image object
             The new image to use.
-
+        ref_image: Image object
+            The reference image to use.
+        im_type: str (optional)
+            If given, will also set the output image's type.
+            Use "diff" or "warped", which will be prepended
+            with "Com" if the new image is a "Com..." type image
+            Note that if not given, the output type will remain as "Sci".
         Returns
         -------
         output: Image
             The new Image object. It would not have any data variables or filepath.
         """
+        if ref_image is None:
+            raise ValueError("Must provide a reference image.")
+        if new_image is None:
+            raise ValueError("Must provide a new image.")
+
         output = Image(nofile=True)
 
         # for each attribute, check the two images have the same value
-        for att in ['section_id', 'instrument', 'telescope', 'filter', 'project', 'target']:
-            ref_value = getattr(ref, att)
-            new_value = getattr(new, att)
+        for att in ['instrument', 'telescope', 'project', 'section_id', 'filter', 'target']:
+            ref_value = getattr(ref_image, att)
+            new_value = getattr(new_image, att)
 
             if att == 'section_id':
                 ref_value = str(ref_value)
                 new_value = str(new_value)
-            if ref_value != new_value:
+
+            if att in ['section_id', 'filter', 'target'] and ref_value != new_value:
```

Comment:
OIC - you're only requiring section, filter, and target to be the same.  But, as I noted, we can't require section to be the same.